src/com/epam/javatrainings/annotations/validator/Validator.java

```diff
@@ -1,57 +1,60 @@
 package com.epam.javatrainings.annotations.validator;
 
 import com.epam.javatrainings.annotations.custom_annotations.*;
-import com.epam.javatrainings.annotations.dto.CustomerDto;
+
 
 import java.lang.reflect.Field;
 import java.time.LocalDate;
 import java.time.Period;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-public class Validator {
+public class Validator<T>{
 
-    public void validate(CustomerDto dto) throws IllegalAccessException, NoSuchFieldException {
+    Pattern VALID_EMAIL_ADDRESS_REGEX;
+    Matcher matcher;
+    List<String> errorList = new ArrayList<>();
 
+    public List<String> validate(T dto) throws IllegalAccessException, NoSuchFieldException {
```

Comment:
make the generic only method, not class:
public <T> List<String> validate(T dto) {}